nipoppy/workflows/dataset_status.py

```diff
@@ -0,0 +1,467 @@
+            elif (row[doughnut_cols] > 0).any():
+                reward_emojis.append(broom_emoji)
+
+            # Additional rewards based on quantity
+            # if row[bids_col] > high_throughput:
+            #     reward_emojis.append(bento_box_emoji)
+            # elif row[bids_col] > medium_throughput:
+            #     reward_emojis.append(sushi_emoji)
+
+            status_df.loc[row_obj[0], reward_col_name] = " ".join(reward_emojis)
+
+        return status_df
+
+    def _add_bagel_rewards(
+        self, status_df: pd.DataFrame, status_col_dict: dict, reward_col_name: str
+    ) -> pd.DataFrame:
+        """Add begal rewards column to the status DataFrame."""
+        manifest_col = status_col_dict["manifest"]
+        doughnut_cols = status_col_dict["doughnut"]
+        bagel_cols = status_col_dict["bagel"]
+
+        status_count_cols = [manifest_col] + doughnut_cols + bagel_cols
+        status_df[status_count_cols] = (
+            status_df[status_count_cols].fillna(0).astype(int)
+        )
+
+        for row_obj in status_df.iterrows():
+            row = row_obj[1]
+            reward_emojis = []
+
+            # bagel emoji if at least one participant has been processed
+            # through all pipelines
+            if (row[bagel_cols] > 0).all():
+                reward_emojis.append(bagel_emoji)
+
+            # confetti_emoji if processing is complete for all
+            # participants and pipelines
+            all_equal = False
+            if (row[status_count_cols] > 0).all():
+                row_vals = row[status_count_cols].values
+                all_equal = all([x == row_vals[0] for x in row_vals])
+                if all_equal:
+                    reward_emojis.append(confetti_emoji)
+
+            # Cannot nest this in prior if statement because
+            # some pipeline count might be zero
+            # party_popper_emoji if at least one pipeline is complete
+            if (not all_equal) & (row[bagel_cols] == row[manifest_col]).any():
+                reward_emojis.append(party_popper_emoji)
+            # rocket_emoji if at least one pipeline is in progress
+            elif (not all_equal) & (row[bagel_cols] > 0).any():
+                reward_emojis.append(rocket_emoji)
+
+            # Additional rewards based on quantity
+            # if (row[bagel_cols] > high_throughput).any():
+            #     reward_emojis.append(mate_emoji)
+            # elif (row[bagel_cols] > medium_throughput).any():
+            #     reward_emojis.append(shaved_ice_emoji)
+
+            status_df.loc[row_obj[0], reward_col_name] = " ".join(reward_emojis)
+
+        return status_df
+
+    def _df_to_table(self, status_df: pd.DataFrame, legend_dict: dict):
+        """Convert a pandas.DataFrame obj into a rich.Table obj."""
+        df = status_df.copy().reset_index()
+        df = df.sort_values(by=self.manifest.col_session_id)
+
+        console = Console()
+
+        crossed_fingers_emoji = "🤞"
```

Comment:
I have had people ask what the crossed fingers meant, so this might be confusing for some people